statsmodels/regression/_stepwise.py

```diff
@@ -0,0 +1,312 @@
+    def params(self):
+        '''current parameter estimates of endog on included exog
+
+        view
+        '''
+        # rows for all endog, cols for included exog
+        ret = self.rs_current[-self.k_vars_y:, self.is_exog]
+        return ret
+
+    @property
+    def df_resid(self):
+        k_incl = self.is_exog.sum()
+
+        return self.nobs - k_incl - self.ddof_model
+
+    @property
+    def scale2(self):
+        return self.rs_current[-1, -1] / self.df_resid
+
+    @property
+    def bse(self):
+        '''current parameter estimates of endog on included exog
+
+        view
+        '''
+        ret = self.normalized_cov_params
+        ret *= self.scale2
+        ret = np.sqrt(ret)
+        return ret
+
+    @property
+    def normalized_cov_params(self):
+        return self.rs_current[self.is_exog, self.is_exog]
+
+    def sweep(self, k, update=True):
+        '''sweep variable k
+
+        Parameters
+        ----------
+        k : int
+            index of the variable to be swept
+        update : bool
+            If True (default), then the attributes are updated. If False,
+            then the instance stays unchanged and the new matrix is returned.
+
+        Returns
+        -------
+        None or rs_next
+            see update parameter
+
+        Notes
+        -----
+        updates is mostly for checking results, we should get most of the
+        results without full updating. See the rss_diff, params_new methods.
+
+        '''
+        # sweep function creates a copy
+        rs_next = sweep(k, self.rs_current)
+        if update:
+            self.rs_current = rs_next
+            self.is_exog[k] = ~self.is_exog[k]  # flip switch
+            self.update_history()
+        else:
+            return rs_next
+
+    def rss_diff(self, endog_idx=-1):
+        '''change in rss when one of a variable is swept
+
+        for now:
+            look only at the effect on a single endog
+            only case endog_idx=-1
+        '''
+        rr = self.rs_current
+        return (rr[-1, :-1]**2 / np.diag(rr)[:-1])
+
+    def ftest_sweep(self):
+        '''get f_test for sweeping a variable
+
+        Warning: currently for adding only
+        incomplete vectorization
+
+        '''
+        from scipy import stats  # lazy for now
+
+        # need to make the following conditional on is_endog, add or drop
+        ssr_diff = self.rss_diff
+        ssr_full = self.rss + ssr_diff
+        df_full = self.df_resid + 1  # add one variable
+        df_diff = 1
+
+        f_value = ssr_diff / df_diff / ssr_full * df_full
+        p_value = stats.f.sf(f_value, df_diff, df_full)
+        return f_value, p_value
+
+    def params_new(self, endog_idx=-1):
+        '''new parameters when one variable is swept
+
+        for now:
+            look only at the effect on a single endog
+            only case endog_idx=-1
+        '''
+        rr = self.rs_current
+        return (rr[-1, :-1] / np.diag(rr)[:-1, None])
```

Comment:
this computes for all variables except endog at index -1
It is not restricted to variables that are not yet in exog. (If we sweep an already swept variable, then we reverse the sweep, according to some references.)

I think this is 2-dim, how all parameters change for sweeping each variable.
columns are parmeters for a single regression
rows are the variables that are swept